### DISCUSSION TITLE
How to CleanUp Resources Created by Executing Custodian Policies

### DISCUSSION THREAD

**Discussion by XuejiaoZhang** [cloud-custodian/cloud-custodian#7389] [General]
Question 1: When executing Custodian policies, it may create resources for example AWS Lambda/AWS Config Rule.
I am wondering what the best practice of cleaning up resources created through executing Custodian policies is.

Question 2: when we run Policies files, what are the expected behaviors of the resource creation/update process?? 
Say we have a Custodian Policies yaml file, including:
policy 1: it will create a Lambda (Lambda 1)
policy 2: it will create a Lambda (Lambda 2)
policy 3: it will create a Lambda (Lambda 3 and AWS Config Rule A)

The first time of running the policies, it will create all the resources (Lambda 1,2,3 and Config Rule A) from scratch.

If I modified the policy 2 without changing the policy name and run the policies again, 
will only Lambda 2 be updated (not delete the old one then create a new one),  the other resources remains the same?

The same If I modified the policy 3 without changing the policy name and run the policies again, 
will only Lambda 3 and Config Rule A be updated (not delete the old one then create a new one),  the other resources remains the same?

What happened to my test is that when modified the policy 2 without changing the policy name and run the policies again, Lambda 2 is updated. But Last modified time of Lambda 1 is changed. 

Would you please give some clue? Thanks.

**Comment by kapilt** [comment]
we have a script in the repo that can cleanup provisioned resources based on whats no longer present in the config file, see readme here https://github.com/cloud-custodian/cloud-custodian/tree/master/tools/ops

**Comment by XuejiaoZhang** [comment]
Hi @kapilt , thanks for your reply.

I intent to test whether AWS Config Rule will be deleted along with Lambda, but met some issue.
Following the link you shared, I run the command below with some of policies commented( no longer present), but  the  Lambda functions were not deleted.
```
$ python cloud-custodian/tools/ops/mugc.py -c demo/custodian_polices.yml     -v
2020-10-21 14:13:20,588: custodian.cache:DEBUG Disabling cache 
```

Do you have any clue on this?

**Comment by kapilt** [comment]
where you targeting the correct region and account? do your lambda function policies have a prefix, running --help will show additional options. also would recommend --dryrun when testing syntax
